src/asr+nmt+tts.js

```diff
@@ -0,0 +1,99 @@
+import serviceId from "../config/serviceId.js";
+import axios from "axios";
+import { authorizationToken } from "../config/verification.js";
+
+
+async function asr_nmt_tts(sourceLang, targetLang, base64) {
+    const url = 'https://dhruva-api.bhashini.gov.in/services/inference/pipeline'
+    let headers;
+    if (authorizationToken) {
+        headers = {
+            'Authorization': authorizationToken,
+
+            'Content-Type': 'application/json',
+            'Access-Control-Allow-Origin': '*'
+        }
+    } else {
+        console.log('no authentication detail passed')
+        return 'no authentication detail passed'
+    }
+    console.log('base64 from package:', base64)
+    console.log(typeof base64)
+
+    const payload = {
+        "pipelineTasks": [
+            {
+                "taskType": "asr",
+                "config": {
+                    "language": {
+                        "sourceLanguage": sourceLang
+                    },
+                    "serviceId": await serviceId('asr', sourceLang),
+                    "audioFormat": "flac",
+                    "samplingRate": 16000
+                }
+            },
+            {
+                "taskType": "translation",
+                "config": {
+                    "language": {
+                        "sourceLanguage": sourceLang,
+                        "targetLanguage": targetLang
+                    },
+                    "serviceId": await serviceId('translation', sourceLang, targetLang)
+                }
+            },
+            {
+                "taskType": "tts",
+                "config": {
+                    "language": {
+                        "sourceLanguage": targetLang
+                    },
+                    "serviceId": await serviceId("tts", targetLang),
+                    "gender": "female",
+                    "samplingRate": 8000
+                }
+            }
+        ],
+        "inputData": {
+            "audio": [
+                {
+                    "audioContent": base64
+                }
+            ]
+        }
+    }
+    return await axios.post(url, payload, { headers }).then(function (res) {
```

Comment:
the setting of headers, the post method etc should be in a base class. This code seems to be repeating in all the js files.